server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -1107,15 +1125,59 @@ public void applyClusterState(final ClusterChangedEvent event) {
         ingestClusterStateListeners.forEach(consumer -> consumer.accept(state));
 
         IngestMetadata newIngestMetadata = state.getMetadata().custom(IngestMetadata.TYPE);
-        if (newIngestMetadata == null) {
-            return;
+        if (newIngestMetadata != null) {
+            try {
+                innerUpdatePipelines(newIngestMetadata);
+            } catch (ElasticsearchParseException e) {
+                logger.warn("failed to update ingest pipelines", e);
+            }
         }
 
-        try {
-            innerUpdatePipelines(newIngestMetadata);
-        } catch (ElasticsearchParseException e) {
-            logger.warn("failed to update ingest pipelines", e);
+        updatePluginPipelines(event);
```

Comment:
This should run on every mapping update? Are we sure that this event will be apply to all nodes?

Reply:
> This should run on every mapping update?

I've been discussing with Armin and we're checking an additional approach, to use a separate metadata structure instead of IndexMetadata.

Otherwise, we need to iterate on all the IndexMetadata in the cluster, which can have a huge number of indices, on every mapping update.

> Are we sure that this event will be apply to all nodes?

In my manual testing, I've tested a master node, a coordinator node, and a data node. They all receive the mapping update.